File: ILT.IHR.APP/ClientApp/src/app/core/interfaces/Leave.ts
import { ILeaveBalanceDisplay } from "./LeaveBalance";

export class ILeave {
    LeaveID: number;
    EmployeeID: number;
    Title: string;
    Detail: string;
    StartDate: Date;
    EndDate: Date;
    IncludesHalfDay: boolean;
    Duration: string;
    LeaveTypeID: number;
    RequesterID: number;
    ApproverID?: number;
    StatusID: number;
    EmployeeName?: string;
    LinkID: string;
    Comment: string;
    CreatedBy: string;
    CreatedDate: Date;
    ModifiedBy: string;
    ModifiedDate: Date;
    TimeStamp: ['']
}
export class ILeaveDisplay {
    leaveID: number;
    employeeID: number;
    employeeName: string;
    requesterID: number;
    requester: string;
    approverID: number;
    approver: string;
    statusID: number;
    statusValue: string;
    status: string;
    leaveTypeID: number;
    leaveType: string;
    title: string;
    comment: string;
    startDate: Date;
    endDate: Date;
    duration: string;
    includesHalfDay: boolean;
    country: string;
    linkID: string;
    detail: string;
    createdBy: string;
    createdDate: Date;
    modifiedBy: string;
    modifiedDate: Date;
    leaveBalances: ILeaveBalanceDisplay[]
}

File: ILT.IHR.DTO/Leave.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ILT.IHR.DTO
{
    public partial class Leave : AbstractDataObject
    {
        public int LeaveID
        {
            get { return base.RecordID; }
            set { base.RecordID = value; }
        }
        [Required]
        public int? EmployeeID { get; set; }
        public string EmployeeName { get; set; }
        [Required]
        public string Title { get; set; }
        public string Detail { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool IncludesHalfDay { get; set; }
        public Decimal Duration { get; set; }
        [Required]
        [RegularExpression("^0*[1-9]\\d*$")]
        public int LeaveTypeID { get; set; }
        public string LeaveType { get; set; }
        public Guid LinkID { get; set; }
        public int RequesterID { get; set; }
        public string Requester { get; set; }
        public int? ApproverID { get; set; }
        public string Approver { get; set; }
        public int StatusID { get; set; }
        public string StatusValue { get; set; }
        public string Status { get; set; }
        public string Comment { get; set; }
        public string Country { get; set; }
        public List<LeaveBalance> LeaveBalances { get; set; }
    }
}